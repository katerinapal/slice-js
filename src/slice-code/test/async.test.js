import {snapSlice} from './helpers/utils'

test(
  'each([1, 2], (i, cb) => cb(), () => {})',
  snapSlice(require.resolve('async/dist/async'), ({each}) => {
    each([1, 2], (i, cb) => cb(), () => {})
  })
)
