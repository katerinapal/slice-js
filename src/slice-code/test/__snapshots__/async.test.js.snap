exports[`test callback(false) && callback(true) 1`] = `
"export { promise, callback };

function promise(pass) {
  return new Promise((resolve, reject) => {
    if (pass) {
      resolve(\'pass\');
    } else {
      reject(\'!pass\');
    }
  });
}

function callback(pass, cb) {
  promise(pass).then(cb, cb);
}"
`;

exports[`test callback(false) 1`] = `
"export { promise, callback };

function promise(pass) {
  return new Promise((resolve, reject) => {
    reject(\'!pass\');
  });
}

function callback(pass, cb) {
  promise(pass).then(cb, cb);
}"
`;

exports[`test callback(true) 1`] = `
"export { promise, callback };

function promise(pass) {
  return new Promise((resolve, reject) => {
    resolve(\'pass\');
  });
}

function callback(pass, cb) {
  promise(pass).then(cb, cb);
}"
`;

exports[`test promise(false) && promise(true) 1`] = `
"export { promise };

function promise(pass) {
  return new Promise((resolve, reject) => {
    if (pass) {
      resolve(\'pass\');
    } else {
      reject(\'!pass\');
    }
  });
}"
`;

exports[`test promise(false) 1`] = `
"export { promise };

function promise(pass) {
  return new Promise((resolve, reject) => {
    reject(\'!pass\');
  });
}"
`;

exports[`test promise(true) 1`] = `
"export { promise };

function promise(pass) {
  return new Promise((resolve, reject) => {
    resolve(\'pass\');
  });
}"
`;
